base/complex.jl

```diff
@@ -114,13 +114,16 @@ real(::Type{Complex{T}}) where {T<:Real} = T
     isreal(x) -> Bool
 
 Test whether `x` or all its elements are numerically equal to some real number,
-`Inf` or `NaN`.
+including infinities and NaNs.
```

Comment:
I think the more correct technical definition would be that `isreal(x)` is true if `isequal(x, real(x))` is true. In other words, the real component of `x` is numerically identical to `x` itself, comparing `NaN`s as equal. You could just add that after your explanation as a more technical clarification.